packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -546,28 +547,37 @@ std::variant<rocksdb::Status, MachineStateKeys> ArbCore::getCheckpointUsingGas(
     marshal_uint256_t(total_gas, key);
     auto key_slice = vecToSlice(key);
     it->SeekForPrev(key_slice);
-    if (!it->Valid()) {
-        if (!it->status().ok()) {
-            return it->status();
+    while (it->Valid()) {
+        if (after_gas) {
+            it->Next();
```

Comment:
This should be outside the loop, and as mentioned below it should continue forwards if after_gas is true. However, I've just noticed that getCheckpointUsingGas is never called with after_gas is true, so I would just cut out this code and remove the parameter.

Reply:
:+1: